### PR TITLE
Updated files for error message fixing

### PR DESCRIPTION
**Pull request by Madhu-NEC**
Fix for issue [IOTA UL- #199](https://github.com/telefonicaid/iotagent-ul/issues/199)

When a measure with unknown/undefined attribute is sent, it gives DEVICE_NOT_FOUND error but it should give ATTRIBUTE_NOT_FOUND error.

### CODE REVIEW

File: lib/errors.js [hunk 1]
```diff
@@ -77,6 +77,11 @@ module.exports = {
         this.message = 'No device was found with id:' + id;
         this.code = 404;
     },
+    AttributeNotFound: function(){
+        this.name = 'ATTRIBUTE_NOT_FOUND';
+        this.message = 'Some of the attributes doesnt exist';
+        this.code = 400;
```
[the remark this answers is not in the document]
Comment (Madhu-NEC) [reply lib/errors.js]: > I think that code is better and the same used by DEVICE_NOT_FOUND (so backward compatibilty is somehow better)

Fixed in https://github.com/telefonicaid/iotagent-node-lib/pull/827/commits/ab2b7ce3e45f4b997d0d20223d37c82be12abf13
Please confirm.

File: lib/errors.js [hunk 2]
```diff
@@ -77,6 +77,11 @@ module.exports = {
         this.message = 'No device was found with id:' + id;
         this.code = 404;
     },
+    AttributeNotFound: function(){
+        this.name = 'ATTRIBUTE_NOT_FOUND';
+        this.message = 'Some of the attributes doesnt exist';
```
[the remark this answers is not in the document]
Comment (Madhu-NEC) [reply lib/errors.js]: Fixed in https://github.com/telefonicaid/iotagent-node-lib/pull/827/commits/ab2b7ce3e45f4b997d0d20223d37c82be12abf13

**Comment by fgalan** [comment]
Just to be sure if I'm understanding this PR correctly... what it does is to provide a more correct log trace message. I mean, the behavior is the same that before (print a trace in the log) but now that trace is describing more precisesly the cause of the problem.

Is my understanding correct? Or is the PR doing something else? Thanks!

**Comment by Madhu-NEC** [comment]
Hi @fgalan ,

> Is my understanding correct? Or is the PR doing something else? Thanks!

Yes, this PR is to provide a more correct log trace message. Previously, API responses 404 DEVICE_NOT_FOUND in case any attribute is not found. Now it will give 404 ATTRIBUTE_NOT_FOUND.

**Comment by Madhu-NEC** [comment]
Hi @AlvaroVega ,
I have resolved the conflict for CHANGES_NEXT_RELEASE.
If the PR is OK then please merge the PR into master.
Thank you.

**Pull request merged**